network/p2p/dnsaddr/resolve.go

```diff
@@ -30,13 +30,19 @@ func isDnsaddr(maddr multiaddr.Multiaddr) bool {
 }
 
 // Iterate runs through the resolvable dnsaddrs in the tree using the resolveController and invokes f for each dnsaddr node lookup
-func Iterate(initial multiaddr.Multiaddr, controller *MultiaddrDNSResolveController, f func(dnsaddr multiaddr.Multiaddr, entries []multiaddr.Multiaddr) error) error {
+func Iterate(initial multiaddr.Multiaddr, controller ResolveController, f func(dnsaddr multiaddr.Multiaddr, entries []multiaddr.Multiaddr) error) error {
 	resolver := controller.Resolver()
 	if resolver == nil {
 		return errors.New("passed controller has no resolvers Iterate")
 	}
+	const maxHops = 100
```

Comment:
Why is this constant set to 100 vs say 10 or 50 or any other number?

Reply:
it is basically any number to have the loop below to terminate eventually. Either 10, 50, 100 are equally good.